exporter/trace/zipkin/zipkin.go

```diff
@@ -0,0 +1,194 @@
+// 		"go.opencensus.io/trace/adaptor/zipkin"
+// 	)
+//	...
+//		localEndpoint, err := zk.NewEndpoint("my server", myHostPort)
+// 		if err != nil {
+// 			log.Print(err)
+// 		}
+// 		exporter, err := zipkin.NewExporter("http://localhost:9411/api/v2/spans", localEndpoint)
+// 		if err != nil {
+// 			log.Print(err)
+// 		} else {
+// 			trace.RegisterExporter(exporter)
+// 		}
+package zipkin
+
+import (
+	"encoding/binary"
+	"strconv"
+	"strings"
+
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter"
+	"github.com/openzipkin/zipkin-go/reporter/http"
+	"go.opencensus.io/trace"
+)
+
+// Exporter is an implementation of trace.Exporter that uploads spans to a
+// Zipkin server.
+type Exporter struct {
+	reporter      reporter.Reporter
+	localEndpoint *model.Endpoint
+}
+
+// NewExporter returns an implementation of trace.Exporter that uploads spans
+// to a Zipkin server.
+//
+// zipkinURL is the address of the Zipkin server.
+//
+// localEndpoint sets the local endpoint of exported spans.  It can be
+// constructed with github.com/openzipkin/zipkin-go.NewEndpoint, e.g.:
+// 	localEndpoint, err := NewEndpoint("my server", listener.Addr().String())
+// localEndpoint can be nil.
+//
+// opts can be used to pass options to the underlying Zipkin http reporter
+// library.
+func NewExporter(zipkinURL string, localEndpoint *model.Endpoint, opts ...http.ReporterOption) (*Exporter, error) {
+	return &Exporter{
+		reporter:      http.NewReporter(zipkinURL, opts...),
+		localEndpoint: localEndpoint,
+	}, nil
+}
+
+// Export exports a span to a Zipkin server.
+func (e *Exporter) Export(s *trace.SpanData) {
+	e.reporter.Send(zipkinSpan(s, e.localEndpoint))
+}
+
+const (
+	statusCodeTagKey        = "census.status_code"
+	statusDescriptionTagKey = "census.status_description"
+)
+
+var sampledTrue = true
+
+func convertTraceID(t trace.TraceID) model.TraceID {
+	return model.TraceID{
+		High: binary.BigEndian.Uint64(t[:8]),
+		Low:  binary.BigEndian.Uint64(t[8:]),
+	}
+}
+
+func convertSpanID(s trace.SpanID) model.ID {
+	return model.ID(binary.BigEndian.Uint64(s[:]))
+}
+
+func spanKind(s *trace.SpanData) model.Kind {
+	if s.HasRemoteParent {
+		return model.Server
+	}
+	if strings.HasPrefix(s.Name, "Sent.") {
+		return model.Client
+	}
+	if strings.HasPrefix(s.Name, "Recv.") {
+		return model.Server
+	}
+	if len(s.MessageEvents) > 0 {
+		switch s.MessageEvents[0].EventType {
+		case trace.MessageEventTypeSent:
+			return model.Client
+		case trace.MessageEventTypeRecv:
+			return model.Server
+		}
+	}
+	return model.Undetermined
+}
+
+func zipkinSpan(s *trace.SpanData, localEndpoint *model.Endpoint) model.SpanModel {
+	sc := s.SpanContext
+	z := model.SpanModel{
+		SpanContext: model.SpanContext{
+			TraceID: convertTraceID(sc.TraceID),
+			ID:      convertSpanID(sc.SpanID),
+			Sampled: &sampledTrue,
+		},
+		Kind:          spanKind(s),
+		Name:          s.Name,
+		Timestamp:     s.StartTime,
+		Shared:        false,
```

Comment:
unless required, you can leave this out as shared defaults to false and json will be smaller.

Reply:
Actually it will not have effect on the JSON payload because the `omitempty` Go tag is set and false is the boolean default/zero value.

I do like it being mentioned here so it is explicit that OpenCensus does not support Zipkin's shared span feature.

Reply:
Yes this line is a no-op which I left in to be explicit.